examples/2d/pixel_grid_snap_2d.rs

```diff
@@ -0,0 +1,158 @@
+//! Shows how to achieve snapping graphics to the pixel grid using rendering to a texture in 2D
+
+use std::f32::consts::PI;
+
+use bevy::{
+    prelude::*,
+    render::{
+        camera::RenderTarget,
+        render_resource::{
+            Extent3d, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages,
+        },
+        view::RenderLayers,
+    },
+};
+use bevy_internal::sprite::MaterialMesh2dBundle;
+use bevy_internal::window::WindowResized;
+
+const RES_WIDTH: u32 = 160;
+const RES_HEIGHT: u32 = 90;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest()))
+        .insert_resource(Msaa::Off)
+        .add_startup_systems((setup_camera, setup_sprite, setup_mesh))
+        .add_systems((transform_drawables, fit_canvas))
+        .run();
+}
+
+#[derive(Component)]
+struct Canvas;
+
+#[derive(Component)]
+struct InGameCamera;
+
+#[derive(Component)]
+struct OuterCamera;
+
+#[derive(Component)]
+struct Drawable;
+
+fn setup_sprite(mut commands: Commands, asset_server: Res<AssetServer>) {
+    let texture = asset_server.load("textures/rpg/chars/sensei/sensei.png");
+    // the sample sprite that will be rendered to the canvas
+    commands.spawn((
+        SpriteBundle {
+            texture: texture.clone(),
+            transform: Transform::from_xyz(-40., 20., 0.),
+            ..default()
+        },
+        Drawable,
+        RenderLayers::layer(1),
+    ));
+
+    // same, but skips the pixel-perfect rendering
+    commands.spawn((
+        SpriteBundle {
+            texture,
+            transform: Transform::from_xyz(-40., -20., 0.),
+            ..default()
+        },
+        Drawable,
+    ));
+}
+
+fn setup_mesh(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    commands.spawn((
+        MaterialMesh2dBundle {
+            mesh: meshes.add(Mesh::from(shape::Quad::default())).into(),
+            transform: Transform::from_xyz(40., 0., 0.).with_scale(Vec3::splat(32.)),
+            material: materials.add(ColorMaterial::from(Color::BLACK)),
+            ..default()
+        },
+        RenderLayers::layer(1),
+        Drawable,
+    ));
+}
+
+fn setup_camera(mut commands: Commands, mut images: ResMut<Assets<Image>>) {
+    let size = Extent3d {
+        width: RES_WIDTH,
+        height: RES_HEIGHT,
+        ..default()
+    };
+
+    // the "canvas"
+    let mut canvas = Image {
+        texture_descriptor: TextureDescriptor {
```

Comment:
Comments on this setup about how/why these setting work would greatly increase the educational value here.

Reply:
Done, added more details to the comments